### ISSUE TITLE
Starting Firebird service fails with "Security database error!" in systemctl status

### ISSUE THREAD

**Issue by hhfiddelke** [FirebirdSQL/firebird#7127]
After installing Firebird 4.0.1 from source on SUSE SLES 15.3 i got the above error message when starting the service.
Then i tried to start fbguard -deamon as root which worked fine. 
After this i changed the service definition file firebird. service to run the service as root, which also worked fine when starting. 
So this had to be a permission issue, but i found no difference in firebird directories compared to other installations.

While searching for filenames used in firebird, i found the -d (debug) switch. Then i changed the firebird user temporary to allow login. With `su - firebird -c "/usr/local/firebird/bin/firebird -d"` i got the hint i needed. It showed me that firebird was not able to access the '/tmp/firebird'- directory that i created as good admin root during installation to contain my installation files, but i created the directory with user root:root not firebird:firebird as required.

So this is a short explanation of my odyssey locating the cause for my problem, and probably a hint to others locating similar issues.

I would request for a future release a more detailed/better explaining error message, that helps a little bit moreand does not show into the wrong direction.

**Comment by AlexPeshkoff** [comment]
On 2/9/22 10:33, Hans H. Fiddelke wrote:

> I would request for a future release a more detailed/better explaining 
> error message, that helps a little bit moreand does not show into the 
> wrong direction.
>

Did you check firebird.log file?

**Comment by hhfiddelke** [comment]
Yes, it only returned that starting the service returns error(2)

**Comment by AlexPeshkoff** [comment]
On 2/9/22 11:13, Hans H. Fiddelke wrote:
>
> Yes, it only returned that starting the service returns error(2)
>

Very strange. The fragment of code:

         // first tries to log error
     gds__log_status(path, status);

         // next tries to type it on a tty
     gds__put_error(path);
     gds__print_status(status);

         // finally puts minimal error string to syslog (you've seen it)
     Firebird::Syslog::Record(Firebird::Syslog::Error, "Security 
database error");

         // exits with status 2 (was logged)
     exit(STARTUP_ERROR);


I understand why you did not see message on tty w/o '-d' switch, but why 
did gds__log_status fail? What about /usr/local/firebird & 
/usr/local/firebird/firebird.log permissions?

**Comment by hhfiddelke** [comment]
Don't know if this is expected, but perhaps we see the messages, but we can't read them, because they are written black on black😉😎
There are several empty lines in the log. Is this an expected behavior?

**Comment by AlexPeshkoff** [comment]
On 2/9/22 14:55, Hans H. Fiddelke wrote:
>
> Don't know if this is expected, but perhaps we see the messages, but 
> we can't read them, because they are written black on black😉😎
>

FB never controls color in the log.

> There are several empty lines in the log. Is this an expected behavior?
>

Sooner of all not.

**Comment by hhfiddelke** [comment]
So is it possible that the log function writes empty log-strings, instead of the values passed to the function, or are the values empty when passed ?
